apps/cms/lib/helpers.ex

```diff
@@ -141,18 +155,18 @@ defmodule CMS.Helpers do
   # Reusable paragraphs are not directly renderable since they act as instance containers.
   # However, these instances can be unpublished. If unpublished, stop and return false.
   # If published, continue checking the nested child paragraph for publish status.
-  defp show_paragraph?(%{"field_reusable_paragraph" => [child]} = parent, query_params) do
+  defp show_paragraph?(%{"field_reusable_paragraph" => [child]} = parent, preview_mode) do
     %{"status" => [parent_status]} = parent
     %{"paragraphs" => [paragraph]} = child
 
     case parent_status do
       %{"value" => false} -> false
-      _ -> show_paragraph?(paragraph, query_params)
+      _ -> show_paragraph?(paragraph, preview_mode)
     end
   end
 
   # In "preview" mode, allow unpublished paragraphs to be rendered if requested
-  defp show_paragraph?(_, %{"preview" => _, "paragraphs" => _}) do
+  defp show_paragraph?(_, preview: true, paragraphs: true) do
```

Comment:
It doesn't seem like these "preview modes" are actually independent of each other: if `preview` is true *and* `paragraphs` is true, this one bit of behavior changes, else it doesn't. Could we simplify to a single boolean?

Reply:
Authors can be in preview mode but request _not_ to see paragraph previews. I kept both elements are part of the match as a safeguard too, to ensure sending just `paragraphs` doesn't expose them (must be in `preview` mode).